### PR TITLE
Feature cmake

### PR DESCRIPTION
**Pull request by jryannel**
The generator is creating a cmake file which builds the plugin and installs it into the build dir import folder.

### CODE REVIEW

**Comment by GordanM** [comment]
@jryannel do you intend to squash these commits? Because the commit messages might need some updates :)

**Comment by jryannel** [comment]
I wil leave the commit messages as is. Too lazy currently

File: qface/builtin/qtcpp/templates/CMakeLists.txt [hunk 1]
```diff
@@ -6,9 +6,11 @@ project({{module_name}})
...
 
 set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_AUTOMOC ON)
 
 set(INSTALL_PATH "${CMAKE_BINARY_DIR}/imports" CACHE STRING "Path where the plugins are deployed")
 
+find_package(Qt5Core REQUIRED)
 find_package(Qt5Qml REQUIRED)
 
 set (HEADERS
```
Comment (jryannel) [inline qface/builtin/qtcpp/templates/CMakeLists.txt]: Do we need the HEADERS still?

Comment (GordanM) [inline qface/builtin/qtcpp/templates/CMakeLists.txt]: Valid point! Fixed now!

**Pull request merged**